### PR TITLE
docs: explain lock-based updates strategy

### PR DESCRIPTION
**Pull request by lucab**
This adds documentation for updates strategies. In particular this
explains how the lock-based strategy works, and how to configure it.

### CODE REVIEW

File: docs/usage/auto-updates.md [hunk 1]
```diff
@@ -35,6 +35,26 @@ It is recommended to setup and monitor canary nodes, but otherwise normal worker
 
 The default and recommended configuration does not set any static wariness value on Zincati side, leaving rollout decisions to Cincinnati backend.
 
+## Strategies for updates finalization
+
+Zincati actively tries to detect and stage new updates whenever they become available.
+Once a new payload has been locally staged, a machine reboot is required in order to atomically apply the update to the system as a whole.
+
+Rebooting a machine does affect any workloads running on it at that time, and can potentially impact services across a whole cluster of nodes.
```
Comment (ashcrow) [inline docs/usage/auto-updates.md]: nit: "Rebooting a machine affects workfoads running on the machine at that time, and ..."

File: docs/usage/auto-updates.md [hunk 2]
```diff
@@ -35,6 +35,26 @@ It is recommended to setup and monitor canary nodes, but otherwise normal worker
...
+For such reason, Zincati allows the user to control when a node is allowed to reboot to finalize an auto-update.
+
+The following finalization strategies are currently available:
+ * immediately reboot to apply an update, as soon as it is downloaded and staged locally (`immediate` strategy, see [relevant documentation][strategy-immediate])
+ * use an external lock-manager to reboot a fleet of machines in coordinated way (`fleet_lock` strategy, see [relevant documentation][strategy-fleet_lock])
```
Comment (jlebon) [inline docs/usage/auto-updates.md]: Minor: in a coordinated way

File: docs/usage/updates-strategy.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+In short, it consists of two operations:
+ * lock: before rebooting, a reboot slot must be locked (and confirmed) by the lock-manager
+ * unlock: after rebooting, any reboot slot owned by the node must be unlocked (and confirmed) by the lock-manager before proceeding further
+
+This protocol is not coupled to any specific backend, and can be implemented on top of any suitable database.
+As an example, [airlock] is a free-software project which implements such protocol on top of [etcd3].
+
+For configuration purposes, such strategy is labeled `fleet_lock` and takes the following configuration parameters:
+ * `base_url` (string, mandatory, non-empty): the base URL for the FleetLock service
```
Comment (jlebon) [inline docs/usage/updates-strategy.md]: Missing trailing period

File: docs/usage/updates-strategy.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+This strategy can be enabled via a configuration snippet like the following:
+
+```toml
+[updates]
+strategy = "immediate"
+```
+
+The `immediate` strategy is an aggressive finalization method which is biased towards finalizing updates as soon as possible, and it is only aware of node-local state.
+
+Such approach is only recommended for environments where temporary service interruption are not problematic, or there is no need for more complex reboot scheduling.
```
Comment (jlebon) [inline docs/usage/updates-strategy.md]: Such an approach

File: docs/usage/updates-strategy.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+[updates]
+strategy = "fleet_lock"
+
+[updates.fleet_lock]
+base_url = "http://example.com/fleet_lock/"
+```
+
+The `fleet_lock` strategy is a conservative method which is biased towards avoiding service disruptions, but it requires an external component which is aware of cluster-wide state.
+
+Such approach is only recommended where nodes are already grouped into an orchestrated cluster, which can thus provide better overall scheduling decisions.
```
Comment (jlebon) [inline docs/usage/updates-strategy.md]: Such an approach

**Comment by lucab** [comment]
Thanks for proofreading, I've applied all suggestions. I'm going to hold this until the implementation at https://github.com/coreos/zincati/pull/121 is also stamped, then merge both.

**Pull request merged**